contracts/mainnet/connectors/pooltogether/main.sol

```diff
@@ -0,0 +1,203 @@
+     * @param user The user to claim tokens for
+     * @param setId Set claimed amount at this ID in `InstaMemory` Contract.
+    */
+    function claim (
+        address tokenFaucet,
+        address user,
+        uint256 setId
+    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
+        TokenFaucetInterface tokenFaucetContract = TokenFaucetInterface(tokenFaucet);
+
+        uint256 claimed = tokenFaucetContract.claim(user);
+
+        setUint(setId, claimed);
+
+        _eventName = "LogClaim(address,address, uint256, uint256)";
+        _eventParam = abi.encode(address(tokenFaucet), address(user), claimed, setId);
+    }
+
+    /**
+     * @dev Runs claim on all passed comptrollers for a user.
+     * @notice Runs claim on all passed comptrollers for a user.
+     * @param tokenFaucetProxyFactory The TokenFaucetProxyFactory address
+     * @param user The user to claim tokens for
+     * @param tokenFaucets The tokenFaucets to call claim on.
+    */
+    function claimAll (
+        address tokenFaucetProxyFactory,
+        address user,
+        TokenFaucetInterface[] calldata tokenFaucets
+    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
+        TokenFaucetProxyFactoryInterface tokenFaucetProxyFactoryContract = TokenFaucetProxyFactoryInterface(tokenFaucetProxyFactory);
+
+        tokenFaucetProxyFactoryContract.claimAll(user, tokenFaucets);
+
+        _eventName = "LogClaimAll(address,address,TokenFaucetInterface[])";
+        _eventParam = abi.encode(address(tokenFaucetProxyFactory), address(user), tokenFaucets);
+    }
+
+    /**
+     * @dev Deposit into Pod
+     * @notice Deposit assets into the Pod in exchange for share tokens
+     * @param prizePool PrizePool address to deposit to
+     * @param pod Pod address to deposit to
+     * @param to The address that shall receive the Pod shares
+     * @param tokenAmount The amount of tokens to deposit.  These are the same tokens used to deposit into the underlying prize pool.
+     * @param getId Get token amount at this ID from `InstaMemory` Contract.
+     * @param setId Set token amount at this ID in `InstaMemory` Contract.
+    */
+    function depositToPod(
+        address prizePool,
+        address pod,
+        address to,
+        uint256 tokenAmount,
+        uint256 getId,
+        uint256 setId
+    ) external payable returns ( string memory _eventName, bytes memory _eventParam) {
+        uint _tokenAmount= getUint(getId, tokenAmount);
+
+        PrizePoolInterface prizePoolContract = PrizePoolInterface(prizePool);
+        address prizePoolToken = prizePoolContract.token();
+
+        PodInterface podContract = PodInterface(pod);
+
+        // Approve pod
+        TokenInterface tokenContract = TokenInterface(prizePoolToken);
+        tokenContract.approve(pod, _tokenAmount);
+
+        podContract.depositTo(to, _tokenAmount);
+
+        setUint(setId, _tokenAmount);
```

Comment:
Since we are going to receive pods share when we deposit, we set share amount in setId instead of `_tokenAmount`. Like this:
```suggestion
        uint256 _podShare = podContract.depositTo(to, _tokenAmount);

        setUint(setId, _podShare);
```

And we can `_podShare` amount in event too.